### PR TITLE
02 css ecommerce product details

### PR DESCRIPTION
**Pull request by mmusk**
Added single product details page

### CODE REVIEW

**Review by coder-mezab — APPROVED** [review]

**Review by anask601 — APPROVED** [review]

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    <title>All Products - RedStore</title>
+    <link rel="stylesheet" href="style.css" />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700&display=swap"
+      rel="stylesheet"
+    />
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.6.0/font/bootstrap-icons.css"
+      integrity="sha384-7ynz3n3tAGNUYFZD3cWe5PDcE36xj85vyFkawcF6tIwxvIecqKvfwLiaFdizhPpN"
```
Comment (narayand16) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:16]: Why do you need integrity attribute here ?

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      crossorigin="anonymous"
+    />
+  </head>
+  <body>
+    <div class="container">
+      <div class="navbar">
+        <div class="logo">
+          <img src="./images/logo.png" width="125px" alt="Logo" />
```
Comment (narayand16) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:24]: please avoid inline style unless its necessary

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            <li><a href="">Home</a></li>
+            <li><a href="">Products</a></li>
+            <li><a href="">About</a></li>
+            <li><a href="">Conact</a></li>
+            <li><a href="">Account</a></li>
+          </ul>
+        </nav>
+        <img
+          src="./images/cart.png"
+          width="30px"
```
Comment (narayand16) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:38]: no inline styles please

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      }
+    </script>
+
+    <!-- JS for Product gallery -->
+    <script>
+      var ProductImg = document.getElementById("ProductImg");
+      var SmallImg = document.getElementsByClassName("small-img");
+
+      SmallImg[0].onclick = function () {
+        ProductImg.src = SmallImg.src[0];
```
Comment (narayand16) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:243]: you should write a generic function for these assignment operations ?

**Review by narayand16 — CHANGES_REQUESTED** [review]

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 1 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: removed @narayand16

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 2 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html:22]: fixed @narayand16

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 3 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html:36]: fixed @narayand16

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 4 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html:241]: Not familiar with JS that much, I just followed the tutorial to add this functionality, will take account of it in future @narayand16

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 5]
```diff
@@ -0,0 +1,254 @@
+<!DOCTYPE html>
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:1]:
```suggestion
file name cannot have space, join them with hyphen
```

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>All Products - RedStore</title>
+    <link rel="stylesheet" href="style.css" />
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:8]: custom style should come last.
how will they know if you wanna use font stylle from google fonts, but they are loaded after your css styles ?

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+  </head>
+  <body>
+    <div class="container">
+      <div class="navbar">
+        <div class="logo">
+          <img src="./images/logo.png" width="125px" alt="Logo" />
+        </div>
+
+        <nav>
+          <ul id="MenuItems">
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:26]: lower case ids and classes

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star"></i>
+          </div>
+          <p>$50.00</p>
+        </div>
+
+        <div class="col-4">
+          <img src="./images/product-3.jpg" />
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:164]: alt ?

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star"></i>
+          </div>
+          <p>$50.00</p>
+        </div>
+        <div class="col-4">
+          <img src="./images/product-3.jpg" />
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:151]: alt ?

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+    <!-- Footer -->
+    <div class="footer">
+      <div class="container">
+        <div class="row">
+          <div class="footer-col-1">
+            <h3>Download Our App</h3>
+            <p>Download App for Android and ios Mobile Phones</p>
+            <div class="app-logo">
+              <img src="./images/play-store.png" />
+              <img src="./images/app-store.png" />
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:187]: alt ?

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+        } else {
+          MenuItems.style.maxHeight = "0px";
+        }
+      }
+    </script>
+
+    <!-- JS for Product gallery -->
+    <script>
+      var ProductImg = document.getElementById("ProductImg");
+      var SmallImg = document.getElementsByClassName("small-img");
```
Comment (faran4engg) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:238]: `camelCase` for variable names !

**Review by faran4engg — CHANGES_REQUESTED** [review]
few minor changes here and there ! once fixed I'll aprove

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+            <i class="bi bi-star-fill"></i>
+            <i class="bi bi-star"></i>
+          </div>
+          <p>$50.00</p>
+        </div>
+      </div>
+    </div>
+
+    <!-- Footer -->
+    <div class="footer">
```
Comment (AlishaPrasad) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:179]: Footer tag

File: tasks-by-users/mmusk/02-second-task-css/product details.html [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+      var MenuItems = documents.getElementById("MenuItems");
+
+      MenuItems.style.maxHeight = "0px";
+
+      function menutoggle() {
+        if (MenuItems.style.maxHeight == "0px") {
+          MenuItems.style.maxHeight = "200px";
+        } else {
+          MenuItems.style.maxHeight = "0px";
+        }
```
Comment (AlishaPrasad) [inline tasks-by-users/mmusk/02-second-task-css/product details.html:231]: Could you convert this if else to ternary

Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html:231]: Not familiar with JS that much, I just followed the tutorial to add this functionality, will take account of it in future @AlishaPrasad

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 6 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: fixed @faran4engg

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 7 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: fixed @faran4engg

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 8 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: added @faran4engg

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 9 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: added @faran4engg

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 10 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: added @faran4engg

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 11 above)
Comment (mmusk) [reply tasks-by-users/mmusk/02-second-task-css/product details.html]: fixed @faran4engg

**Review by shaikh-7abish — APPROVED** [review]

File: tasks-by-users/mmusk/02-second-task-css/product details.html (hunk 12 above)
Comment (faran4engg) [reply tasks-by-users/mmusk/02-second-task-css/product details.html:183]: @mmusk  where is this resolved ?

**Review by AbbasShaikh0304 — APPROVED** [review]

**Review by mujeeb100 — APPROVED** [review]

**Comment by Abbas4tech** [comment]
Is this PR intended to be merge? @mmusk

**Pull request merged**